src/configure.ac

```diff
@@ -3,8 +3,18 @@ AC_INIT(reaver, 1.5)
 AC_PROG_CC
 AC_LANG(C)
 
+
 CFLAGS="-Wall $CFLAGS"
-LDFLAGS="-ldl -lm -lpcap -lsqlite3 $LDFLAGS"
+
+
+if test `uname` = "FreeBSD"; then
+  LDFLAGS="-lpcap -lsqlite3 $LDFLAGS"
```

Comment:
i think `-ldl` is not even needed for linux. just removing it would be much cleaner than this ifdef hell.